image/templates/helm/stackrox-secured-cluster/internal/cluster-config.yaml.tpl

```diff
@@ -2,7 +2,7 @@
 clusterName: {{ ._rox.clusterName }}
 {{- end }}
 managedBy: {{ ._rox.managedBy }}
-notHelmManaged: {{ eq ._rox.managedBy "MANAGER_TYPE_MANUAL" }}
+notHelmManaged: {{ not (eq ._rox.managedBy "MANAGER_TYPE_HELM_CHART") }}
```

Comment:
It seems `notHelmManaged` predates the existence of the operator. So neither value seems obviously correct when installed via operator, especially since operator uses helm.

Can you please explain what is influenced by the value of this setting so we can make an informed decision?

Reply:
It is (in my opinion) clearly an inconsistency, which should be fixed one way or another, because it would allow configuration of the form
```
notHelmManaged: false # i.e. helmManaged: true
managedBy: MANAGER_TYPE_KUBERNETES_OPERATOR
```

There are currently a handful of occurrences of `GetNotHelmManaged()` in the codem, even though the field is already marked as deprecated. I guess it would be even better to migrate away from the usage of this field and then remove it from new cluster-configs.

Reply:
Secured cluster upgrader uses it to decide whether it should run or not. This can be easily migrated to `managedBy`. There is also one place that uses it due to:

> 	// Be backwards compatible for older Helm charts, which do not send the `managedBy` field: derive `managedBy` from the provided `notHelmManaged` property.

so maybe the other places do it as well?

I can open a PR to refactor that and drop the backwards compatibility - I think, we no longer require to be [compatible with 3 year old versions](https://github.com/stackrox/rox/pull/9356).

Reply:
> Secured cluster upgrader uses it to decide whether it should run or not.

So I think we **should not** be making this change. Since we only want the upgrader to touch clusters installed with manifests.

Reply:
➕ 1000 to stop using this field.

Reply:
I allowed myself to add [546c2e5](https://github.com/stackrox/stackrox/pull/12935/commits/546c2e59d05a02fa1a0b5379f17fcd4a595f6265) to this PR.

Reply:
I don't think we should be changing this line. `{{ eq ._rox.managedBy "MANAGER_TYPE_MANUAL" }}` sounds exactly what the intended semantics should be.